ekncontent/ekncontent/eknc-domain.c

```diff
@@ -940,11 +937,13 @@ query_task (GTask *task,
   if (g_task_return_error_if_cancelled (task))
     return;
 
-  g_autoptr(GMutexLocker) db_lock = g_mutex_locker_new (&state->domain->db_lock);
+  g_autofree char *query_str = eknc_query_object_get_corrected_query_string (state->query);
+
+  g_debug (G_STRLOC ": Query: '%s'", query_str);
 
-  g_autoptr(XapianMSet) results =
-    eknc_database_manager_query (state->db_manager, state->params, &error);
+  g_autoptr(XapianMSet) results = NULL;
```

Comment:
Not sure why the split here, but OK

Reply:
Intermediate state between the "run two queries" and the "run a better query" approaches.

I think I should just squash most of the PR.